python_modules/dagster/dagster/_core/definitions/events.py

```diff
@@ -68,11 +71,19 @@ def __new__(cls, asset_key: AssetKey, partitions: Optional[AbstractSet[str]] = N
         return super(AssetLineageInfo, cls).__new__(cls, asset_key=asset_key, partitions=partitions)
 
 
+class EventWithMetadata(ABC):
```

Comment:
In the future, I'm imagining that this would just be a base `Event` type, and all events should have metadata. There could also be other methods like `with_tags` or whatever. So maybe `EventWithMetadata` is too specific?

This is an internal thing though, so naming doesn't seem as important right now, so happy to accept what we have and then we can refactor later.